docs/sphinx/dev/dev_dot_product_example.rst

```diff
@@ -0,0 +1,67 @@
+********************
+Dot Product Example
+********************
+The following example outlines how to create a dot product between two vectors: A and B, using a `kotekan` pipeline.
+
```

Comment:
I would generate a graph of what this pipeline looks like visually.  e.g. with the existing dot system or the new javascript debug tool.  I think it's helpful to get a visual description of the pipeline along with the written info.